src/python/pants/backend/helm/util_rules/chart.py

```diff
@@ -226,21 +250,34 @@ def path(self) -> str:
 @dataclass(frozen=True)
 class HelmChartRequest:
     field_set: HelmChartFieldSet
+    generate_chart_lockfile: bool = False
```

Comment:
Adding some help for this property would be useful... this will generate the lockfile within the sandbox, but not commit to disk, presumably?

There is some infrastructure for generic lockfile generation to plug into the `generate-lockfiles` goal, if that's relevant here. As it stands, if a lockfile is stale, the idea would be to error and ask someone to run `generate-lockfiles` manually... but our hope is to be able replace it with an automatic facility via https://github.com/pantsbuild/pants/issues/12014.

Reply:
this is a tricky one. In a way I would like to implement a lockfile approach as other backends do, one that hooks into `generate-lockfiles`. And similarly, users could define all the third party artifacts under `3rdparty/helm/...` so that they can manage all potential Helm dependencies in one common place.

I need to admit this implementation here is a workaround to help the `package` goal generate a chart-specific `Chart.lock` file when creating the built package.

Maybe worth removing it for now (it isn't crucial) and address the whole lockfile thing in a future PR

Reply:
I agree that isolating integration with `generate-lockfiles` into a new PR would be a good idea, yea. There are likely to be edge cases since it has not been used very widely yet.